libraries/node-core-library/src/JsonFile.ts

```diff
@@ -222,11 +268,63 @@ export class JsonFile {
     return true;
   }
 
+  /**
+   * An async version of {@link JsonFile.loadAndValidateWithCallback}.
+   */
+  public static async saveAsync(jsonObject: JsonObject, jsonFilename: string, options?: IJsonFileSaveOptions): Promise<boolean> {
+    if (!options) {
+      options = { };
+    }
+
+    // Do we need to read the previous file contents?
+    let oldBuffer: Buffer | undefined = undefined;
+    if (options.updateExistingFile || options.onlyIfChanged) {
+      try {
+        oldBuffer = await FileSystem.readFileToBufferAsync(jsonFilename);
+      } catch (error) {
+        // Ignore this error, and try writing a new file.  If that fails, then we should report that
+        // error instead.
+      }
+    }
+
+    let jsonToUpdate: string = '';
+    if (options.updateExistingFile && oldBuffer) {
+      jsonToUpdate = oldBuffer.toString();
+    }
+
+    const newJson: string = JsonFile.updateString(jsonToUpdate, jsonObject, options);
+
+    const newBuffer: Buffer = Buffer.from(newJson); // utf8 encoding happens here
+
+    if (options.onlyIfChanged) {
+      // Has the file changed?
+      if (oldBuffer && Buffer.compare(newBuffer, oldBuffer) === 0) {
```

Comment:
Any chance we can get this as a side effect of `JsonFile.updateString`? It's almost certainly easier to compute if anything changed while changes are being made

Reply:
`jju` doesn't seem to provide that information. We can revisit this later.